src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt

```diff
@@ -0,0 +1,69 @@
+package com.hibob.academy.employee_feedback_feature.resource
+
+import com.hibob.academy.employee_feedback_feature.dao.EmployeeRole
+import com.hibob.academy.employee_feedback_feature.dao.FeedbackRequest
+import com.hibob.academy.employee_feedback_feature.dao.FeedbackSubmission
+import com.hibob.academy.employee_feedback_feature.service.FeedbackService
+import jakarta.ws.rs.*
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.core.Context
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+
+@Controller
+@Path("/api/feedback")
+class FeedbackResource(private val feedbackService: FeedbackService) {
+    @Path("/submit")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun submitFeedback(@Context requestContext: ContainerRequestContext, newFeedback: FeedbackRequest): Response {
+        val companyId = requestContext.getProperty("companyId") as? Number
+            ?: return Response.status(Response.Status.BAD_REQUEST).entity("Bad Request: Missing companyId").build()
```

Comment:
In this case, if there's a problem with the cookie - i would return NOT AUTHORIZED or NO PERMISSIONS, not bad request

Reply:
Got it
Fixed